raft/raft_test.go

```diff
@@ -288,33 +288,80 @@ func TestProgressPaused(t *testing.T) {
 }
 
 func TestLeaderElection(t *testing.T) {
```

Comment:
Probably separate the test case into TestLeaderElection and TestLeaderElectionWithPreVote? Same for TestLeaderCycle?

And add a helper function testLeaderElection(prevote bool)?

I do not really have a strong opinion here. just feel it might be cleaner.